ai4good/models/cm/functions.py

```diff
@@ -29,6 +29,131 @@ class Simulator:
     def __init__(self, params: Parameters):
         self.params = params
 
+    def ode_system2d(self, t, y,  # state of system
+                   infection_matrix, age_categories, symptomatic_prob, hospital_prob, critical_prob, beta,  # params
+                   latentRate, removalRate, hospRate, deathRateICU, deathRateNoIcu,  # more params
+                   better_hygiene, remove_symptomatic, remove_high_risk, ICU_capacity  # control
+                   ):
+        ##
+        params = self.params
+
+        y2d = y.reshape(age_categories, self.params.number_compartments).T
+        dydt2d = np.zeros(y2d.shape)
+
+        index_I = params.categories['I']['index']
+        index_S = params.categories['S']['index']
+        index_E = params.categories['E']['index']
+        index_C = params.categories['C']['index']
+        index_Q = params.categories['Q']['index']
+        index_U = params.categories['U']['index']
+        index_H = params.categories['H']['index']
+        index_A = params.categories['A']['index']
+
+        I_vec = y2d[index_I, :]
+        H_vec = y2d[index_H, :]
+        A_vec = y2d[index_A, :]
+        C_vec = y2d[index_C, :]
+        S_vec = y2d[index_S, :]
+
+        E_latent = latentRate * y2d[index_E, :]
+        I_removed = removalRate * I_vec
+        Q_qarantined = params.quarant_rate * y2d[index_Q, :]
```

Comment:
there's a typo in the spelling in quarantined..please correct

Reply:
fixed